File: Test_Product.cpp
#include "Product.cpp"
#include <gtest/gtest.h>
#include <string>

using namespace std;

TEST(ProductTest, PureProductFunctionality){
    Product testProduct;
    ASSERT_EQ(0, testProduct.getPrice());
    ASSERT_NE(FARMPRODUCT, testProduct.getObjectType());
    ASSERT_NE(BISON_MEAT, testProduct.getObjectType());
    testProduct.initializeProduct(20, FARMPRODUCT, BISON_MEAT, '?');
    ASSERT_EQ(20, testProduct.getPrice());
    ASSERT_EQ(FARMPRODUCT, testProduct.getProductType());
    ASSERT_EQ(BISON_MEAT, testProduct.getObjectType());
}

File: Java/src/test/java/engisubs/gameobject/product/TestProduct.java
package engisubs.gameobject.product;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;

import engisubs.gameobject.GameObject.GameObjectType;
import engisubs.gameobject.product.Product.ProductType;

import org.junit.Assert;

public class TestProduct {
    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
    private Product test;

    @Before
    public void setUpStreams() {
        test = new Product();
        System.setOut(new PrintStream(outContent));
    }
    @After
    public void cleanStreams() {
        System.setOut(null);
    }

    @Test
    public void initDefaultObjectTest(){
        Assert.assertEquals(0, test.getPrice());
        test.render();
        Assert.assertEquals("?", outContent.toString());
    }

    @Test
    public void initProductTest(){
        test.initializeProduct(80, ProductType.FARMPRODUCT, GameObjectType.BISON_MEAT, "Name");
        Assert.assertEquals(80, test.getPrice());
        Assert.assertEquals(ProductType.FARMPRODUCT, test.getProductType());
        Assert.assertEquals(GameObjectType.BISON_MEAT, test.getGameObjectType());
        Assert.assertEquals("Name", test.getProductName());
    }
}